superdesk/activity.py

```diff
@@ -147,23 +156,38 @@ def on_update(self, updates, original):
         user_id = str(user.get('_id'))
 
         # make sure that the user making the read notification is in the notification list
-        if user_id not in updates.get('read').keys():
+        if not self.is_recipient(updates, user_id):
             raise SuperdeskApiError.forbiddenError('User is not in the notification list')
 
         # make sure the transition is from not read to read
-        if not (updates.get('read')[user_id] == 1 and original.get('read')[user_id] == 0):
+        if not self.is_read(updates, user_id) and self.is_read(original, user_id):
             raise SuperdeskApiError.forbiddenError('Can not set notification as read')
 
         # make sure that no other users are being marked as read
-        for read_entry in updates.get('read'):
-            if read_entry != user_id:
-                if updates.get('read')[read_entry] != original.get('read')[read_entry]:
+        for recipient in updates.get('recipients'):
```

Comment:
Is the `recipients` field always present? If not, a default value should be passed to `get()` as well (e.g. an empty list) to avoid an error.